docs/rest-api.md

```diff
@@ -0,0 +1,383 @@
+# REST API
+
+Flarum exposes a REST API which is used by the single page application but also available for external scripts.
+
+The API follows the best practices defined by the [JSON:API](https://jsonapi.org/) specification.
+
+## Authentication
+
+The single page app uses session cookies to authenticate against the API.
+External scripts can use stateless authentication using [API Keys](#api-keys) or [Access Tokens](#access-tokens).
+
+`GET` endpoints can be used without authentication.
+Only content visible to guests will be returned.
+Other endpoints generally cannot be used without authentication because of the [CSRF protection](#csrf-protection).
+
+### API keys
+
+API Keys are the primary way for scripts, tools and integrations to interact with Flarum.
+
+#### Creation
+
+There is currently no UI to manage API Keys, but they can be created manually in the `api_keys` table of the database.
+
+The following attributes can be filled:
+
+- `key`: Generate a long unique token (recommended: alpha-numerical, 40 characters) and set it here, this will be the token used in the `Authorization` header.
+- `user_id`: Optional. If set, the key can only be used to act as the given user.
+
+The remaining attributes are either automatically filled or currently not used:
+
+- `id`: Will be filled by MySQL auto-increment.
+- `allowed_ips`: Not implemented.
+- `scopes`: Not implemented.
+- `created_at`: Can be set to any date, but is meant for the date of creation of the key.
+- `last_activity_at`: Will be updated automatically when the token is used.
+
+#### Usage
+
+Attach your key value to each API request using the `Authorization` header.
+Then provide the user ID you want to interact as at the end of the header:
+
+    Authorization: Token yourApiKeyValue; userId=1
```

Comment:
```suggestion
    Authorization: Token YOUR_API_KEY_VALUE; userId=1
```

Just a nitpick, but this format might make it easier to see that this should be replaced by the value.

Reply:
Makes sense. I'll see if there are other comments before batch-committing the changes.